client/ayon_core/plugins/load/export_otio.py

```diff
@@ -0,0 +1,187 @@
+            | QtCore.Qt.WindowMinimizeButtonHint
+        )
+
+        self.setStyleSheet(style.load_stylesheet())
+
+        input_widget = QtWidgets.QWidget(self)
+        input_layout = QtWidgets.QGridLayout(input_widget)
+
+        self._project_name = contexts[0]["project"]["name"]
+
+        representations_by_version_id = defaultdict(list)
+        self._version_by_representation_id = {}
+        all_representation_names = set()
+        self._version_path_by_id = {}
+        for context in contexts:
+            version_id = context["version"]["_id"]
+            version = get_version_by_id(self._project_name, version_id)
+            representations = list(get_representations(
+                self._project_name, version_ids=[version_id]
+            ))
+            representations_by_version_id[version_id] = representations
+
+            for representation in representations:
+                all_representation_names.add(representation["name"])
+                id = representation["_id"]
+                self._version_by_representation_id[id] = version
+
+            self._version_path_by_id[version_id] = "{}/{}/{}/v{:03d}".format(
+                representations[0]["context"]["hierarchy"],
+                representations[0]["context"]["asset"],
+                representations[0]["context"]["subset"],
+                representations[0]["context"]["version"]
+            )
+
+        all_representation_names = sorted(all_representation_names)
+
+        input_layout.addWidget(QtWidgets.QLabel("Representations:"), 0, 0)
+        toggle_all_checkboxes = {}
+        for count, name in enumerate(all_representation_names):
+            checkbox = QtWidgets.QCheckBox(name)
+            input_layout.addWidget(
+                checkbox,
+                0,
+                count + 1,
+                alignment=QtCore.Qt.AlignCenter
+            )
+            toggle_all_checkboxes[name] = checkbox
+            checkbox.stateChanged.connect(self.toggle_all)
+
+        self._representation_checkboxes = defaultdict(list)
+        row = 1
+        items = representations_by_version_id.items()
+        for version_id, representations in items:
+            version_path = self._version_path_by_id[version_id]
+            input_layout.addWidget(QtWidgets.QLabel(version_path), row, 0)
+
+            representations_by_name = {x["name"]: x for x in representations}
+            for count, name in enumerate(all_representation_names):
+                checkbox = QtWidgets.QCheckBox()
+                checkbox.setChecked(False)
+                if name in representations_by_name.keys():
+                    self._representation_checkboxes[name].append(
+                        {
+                            "checkbox": checkbox,
+                            "representation": representations_by_name[name]
+                        }
+                    )
+                else:
+                    checkbox.setEnabled(False)
```

Comment:
If you only ever want one of the Checkboxes to be enabled - then I think what you're looking for is making this checkbox group in the Qt UI so Qt itself handles it for you?

Reply:
There can be multiple checkboxes enabled. Depends on the representations. Is that what you mean?

Reply:
Ah sorry, ignore me then. :) I thought the idea was that for each version the representations were 'exclusive' checkboxes and only one could be selected.

I'm not sure what to expect from an otio where I pick multiple representations though haha ;) and are you sure that works? Since you're currently storing them by "name" in the clips data `dict` where name is the version path - which would not be unique per its representation, and thus in the clips data only would end up being present.

Reply:
Yeah, good point about multiple representations. Not sure about the use case either, so i'll limit it to 1 representation per version.